include/IECore/Object.h

```diff
@@ -200,7 +201,7 @@ class IECORE_API Object : public RunTimeTyped
 				typename T::Ptr copy( const T *toCopy );
 			private :
 				ObjectPtr copyInternal( const Object *toCopy );
-				struct CopiedObjects;
+				struct CopiedObjects : public std::map<const Object *, Object *> {};
```

Comment:
Hmm, this is unfortunate. The entire point of the forward declaration and `unique_ptr` was to avoid exposing the `CopiedObjects` class publicly in the header. What's more this is explicitly required to work by the C++ standard :

> std::unique_ptr may be constructed for an incomplete type T, such as to facilitate the use as a handle in the pImpl idiom.

I think though that I may have botched it by not adding an explicit `~CopyContext` constructor. That might force the compiler to generate an implicit one when it doesn't have access to the full `CopiedObjects` definition, and that would be an error :

> If the default deleter is used, T must be complete at the point in code where the deleter is invoked, which happens in the destructor, move assignment operator, and reset member function of std::unique_ptr. 

Could you try an alternative fix? Revert everything from this commit, add a declaration for `~CopyContext()` right beneath the `CopyContext()` constructor, and then define that as `CopyContext::~CopyContext() {}` in the Object.cpp?